web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,161 @@
+
+## Proposal
+
+Derived from the hierarchical visual in the previous section, this PDEP proposes that pandas supports at least all of the following _logical_ types, excluding any type widths for brevity:
+
+  - Signed Integer
+  - Unsigned Integer
+  - Floating Point
+  - Fixed Point
+  - Boolean
+  - Date
+  - Datetime
+  - Duration
+  - Interval
+  - Period
+  - Binary
+  - String
+  - Dictionary
+  - List
+  - Struct
+
+To ensure we maintain all of the current functionality of our existing type system(s), a base type structure would need to look something like:
+
+```python
+class BaseType:
+
+    @property
+    def dtype_backend -> Literal["pandas", "numpy", "pyarrow"]:
+        """
+        Library is responsible for the array implementation
+        """
+        ...
+
+    @property
+    def physical_type:
+        """
+        How does the backend physically implement this logical type? i.e. our
+        logical type may be a "string" and we are using pyarrow underneath -
+        is it a pa.string(), pa.large_string(), pa.string_view() or something else?
+        """
+        ...
+
+    @property
+    def missing_value_marker -> pd.NA|np.nan:
+        """
+        Sentinel used to denote missing values
+        """
+        ...
+```
+
+The theory behind this PDEP is that most users _should not_ care about the physical type that is being used. But if the abstraction our logical type provides is too much, a user could at least inspect and potentially configure which physical type to use.
+
+With regards to how we may expose such types to end users, there are two currently recognized proposals. The first would use factory functions to create the logical type, i.e. something like:
+
+```python
+pd.Series(["foo", "bar", "baz"], dtype=pd.string()  # assumed common case
+pd.Series(["foo", "bar", "baz"], dtype=pd.string(missing_value_marker=np.nan)
+pd.Series(["foo", "bar", "baz"], dtype=pd.string(physical_type=pa.string_view())
+```
+
+Another approach would be to classes:
+
+```python
+pd.Series(["foo", "bar", "baz"], dtype=pd.StringDtype()
+pd.Series(["foo", "bar", "baz"], dtype=pd.StringDtype(missing_value_marker=np.nan)
+pd.Series(["foo", "bar", "baz"], dtype=pd.StringDtype(physical_type=pa.string_view())
+```
+Note that the class-based approach would reuse existing classes like ``pd.StringDtype`` but change their purpose, whereas the factory function would more explicitly be a new approach. This is an area that requires more discussion amongst the team.
```

Comment:
Given that this is a dtype, I'd prefer having `Dtype` in the name.  That favors the class based approach.  But the factory function approach could be used as well, but I'd prefer `pd.stringDtype()` over `pd.string()` to make it clear that it's a dtype.

Reply:
Cool - I think @jorisvandenbossche has the same preference. So happy to start focus on that pattern more.

The only concern I have is that we are repurposing something that has long existed. `pd.Int64Dtype()` gets a new meaning and may no longer return the pandas extension type we created...but maybe that isn't a bad thing

Reply:
i dont have a strong preference, but do think it will make the discussion easier to focus on just one variant

Reply:
Alright I'll stick with that in the next revision. We can always revisit if that doesn't work out for whatever reason. Thanks for the feedback all

Reply:
To be honest, my _personal_ preference would be to go with the shorter functional constructors (`pd.string()`, `pd.int64()`, etc). 
It's easier to use, less boilerplate, generally more similar to other libraries (this was my impression, but looking up a few examples, that is not necessarily true), and ties less into the exact Dtype class (in case two variants of one logical type want to use a different dtype class, although I am not sure this is necessarily something we _want_, though)

But agreed that for now, I think you can just use one throughout the document to explain everything, and then I would add one section on this topic mentioning some options and why which choice was made (and we can still further bikeshed on the exact syntax, but that feels like an actual bikeshed separate from the core ideas of the PDEP)

(@Dr-Irv if you want "dtype" in the name, yet another alternative would be to put the functions in a namespace, like `pd.dtypes.string()`. I seem to remember this also came up when I discussed topic in Basel last year)

> The only concern I have is that we are repurposing something that has long existed. `pd.Int64Dtype()` gets a new meaning and may no longer return the pandas extension type we created...but maybe that isn't a bad thing

Well, we still need to discuss what those logical type constructors would actually return ;) In my mind, it _could_ continue to return our extension types (as I wouldn't necessarily distinguish user-facing extension types and logical types), and then `pd.Int64Dtype()` would not necessarily have to change return value .. (except that it might gain additional keyword to e.g. get the pyarrow-backed integer dtype)

Reply:
> (@Dr-Irv if you want "dtype" in the name, yet another alternative would be to put the functions in a namespace, like `pd.dtypes.string()`. I seem to remember this also came up when I discussed topic in Basel last year)

That works for me.  I'm not a big fan of continually adding things to the top level `pandas` namespace.